apps/studio/src/features/editing-experience/hooks/useLinkEditorModal.tsx

```diff
@@ -0,0 +1,47 @@
+import type { Editor } from "@tiptap/react"
+import { useDisclosure } from "@chakra-ui/react"
+
+import { LinkEditorModal } from "~/components/PageEditor/LinkEditorModal"
+
+interface UseLinkEditorModalProps {
+  editor: Editor
+}
+
+export const useLinkEditorModal = ({ editor }: UseLinkEditorModalProps) => {
+  const {
+    isOpen: isLinkModalOpen,
+    onOpen: onLinkModalOpen,
+    onClose: onLinkModalClose,
+  } = useDisclosure()
+
+  return {
+    onLinkModalOpen,
+    LinkEditorModal: (
```

Comment:
i don't think this is a common pattern (to return the component) - why this is so is because you're forcing a UI onto your end user. 

i'd rather we define a contract like `LinkEditorModalProps` and this returns a method like `getLinkEditorModalProps` that satisfies the contract. this is done for other libraries like chakra (`getInputProps` for `useRadio`) and rhf (`...register`)

Reply:
Oh yeah hor I think I went a bit crazy here, simplified the original component so it takes in the simpler props similar to the table settings modal.